### ISSUE TITLE
Automatically route services registered in Consul

### ISSUE THREAD

**Issue by far-blue** [kobolog/gorb#33]
This is looking like a very interesting project and a great idea - exposing IPVS via a Rest API! I think it would be very cool if Gorb could monitor services in Consul and use Consul's health checks and then create and destroy routes based on this info. There's a similar project called Fabio (https://github.com/eBay/fabio) which does this for http(s) proxying and it works beautifully. It relies on tags registered with the service in Consul and only routes to services which pass consul's health checks (all proxy'd services are required to have a health check).

I don't know if gorb can do multi-host routing like fabio will but it would be brilliant if it could - it would allow any 'edge' of a cluster to accept incoming connections for a container anywhere in the cluster.

**Comment by kobolog** [comment]
Yep, that's a great idea =) Is this a suggestion or you're willing to lend a hand and create a PR?

**Comment by far-blue** [comment]
Generally, I go for PRs but unfortunately I don't (yet) know Go so for the moment it's a suggestion :) Go is on my todo list...

**Comment by kobolog** [comment]
Go is extremely simple, and I believe that the best way to learn a new language is to write something small but useful – like this PR =) I can help you with it! What's your main language now?

**Comment by albamc** [comment]
I created a #39 that sync data to kvstore such as consul in gorb.
I tried to use consul service information registered from [registrator] (https://github.com/gliderlabs/registrator) at first, but there was the following problem.
- There is a problem when trying to use another kvstore other than consul.
- consul service information does not have enough information for load balancing (ex. VIP, method, ...).
So what I would do is to implement gorb-only data in #39 and add a backend(in registrator it says registry) for gorb to registrator (https://github.com/gliderlabs/registrator) I would like to.
